modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -435,6 +435,8 @@ Error GDScriptAnalyzer::resolve_class_inheritance(GDScriptParser::ClassNode *p_c
 		if (!p_class->extends_path.is_empty()) {
 			if (p_class->extends_path.is_relative_path()) {
 				p_class->extends_path = class_type.script_path.get_base_dir().path_join(p_class->extends_path).simplify_path();
+			} else if (p_class->extends_path.begins_with("uid://")) {
+				p_class->extends_path = ResourceUID::ensure_path(p_class->extends_path);
```

Comment:
```suggestion
				p_class->extends_path = ResourceUID::uid_to_path(p_class->extends_path);
```
`ensure_path()` should only be used when the path can be either path or UID.